types/ibc.go

```diff
@@ -18,6 +18,23 @@ type IBCChannelOpenMsg struct {
 	OpenTry  *IBCOpenTry  `json:"open_try,omitempty"`
 }
 
+// GetChannel returns the IBCChannel in this message.
+func (msg IBCChannelOpenMsg) GetChannel() IBCChannel {
+	if msg.OpenInit != nil {
+		return msg.OpenInit.Channel
+	}
+	return msg.OpenTry.Channel
+}
+
+// GetCounterVersion checks if the message has a counterparty version and
+// returns it if so.
+func (msg IBCChannelOpenMsg) GetCounterVersion() (ver string, ok bool) {
```

Comment:
I did some digging and this seems like the conventional way of returning a value that may or may not be rubbish in Go. Oh well!

Reply:
🤷‍♂️

Reply:
Yup, well done.

Go: where they hide from pointers unless you need to pass something in to be mutated by another function